backend/src/main/java/org/cryptomator/hub/api/ValidationUtil.java

```diff
@@ -0,0 +1,11 @@
+package org.cryptomator.hub.api;
+
+public final class ValidationUtil {
+
+	public static final String ID_PATTERN = "[-\\w]+";
+	public static final String JWX_PATTERN = "[-=.\\w]+";
```

Comment:
Base64Url uses `=` for padding, if needed (see https://datatracker.ietf.org/doc/html/rfc4648#page-7).

Additional, if `.` appears in a group, it loses its special meaning of a match all expression, see https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html?is-external=true, section "Character Classes".

Reply:
> note that JWX use base64url (_-), not base64 (/+=).

Reply:
> Additional, if `.` appears in a group

yes but when you use `.` in a regex, you match *any char*, not `.`. What you want to use is`\.`

Reply:
In a regular expresion, if `.` is used in a character class, it _does not_ macht any char. (since it loses its special meaning inside of it). It only matches the dot character itself.
```
jshell> Pattern.matches("[a.]+", "bb")
$1 ==> false

jshell> Pattern.matches("[a.]+", "a.a")
$3 ==> true
```

Reply:
outdated